GameOfLife.rb

```diff
@@ -0,0 +1,58 @@
+class GameOfLife
+  attr_writer :board
+  attr_reader :width
+  attr_reader :height
+
+  def initialize(width, height)
+    @board = Array.new(width) { Array.new(height) { 0 } }
+    @width = width
+    @height = height
+  end
+
+  def is_alive(x, y)
+    sum = 0
+    for i in x - 1..x + 1
+      for k in y - 1..y + 1
+        if (i != x || k != y) && k >= 0 && i >= 0 && k < @height && i < @width
+          if @board[i][k] == 1
+            sum += 1

```

Comment:
ten kawałek jest zdecydowanie za skomplikowany. Spróbuj porozbijać to na mniejsze predykaty.